Makefile

```diff
@@ -1,20 +1,71 @@
 SHELL := /bin/bash
 .SHELLFLAGS = -euo pipefail -c
-MAKEFLAGS = -s
-CGO_ENABLED = 1
+# MAKEFLAGS = -s
+CGO_ENABLED = 0
 BUILDTAGS :=
-BUILDFLAGS := -ldflags '-linkmode=external'
+BUILD_PATH:=build
+BUILD:=ascli
+ifeq ($(CGO_ENABLED),1)
+BUILDFLAGS := -linkmode=external
+else
+BUILDFLAGS :=
+endif
+GOOS           :=$(shell go env GOOS)
+GOARCH         :=$(shell go env GOARCH)
+GOMODULECMD    :=main
+
+
+SEMVER_VERSION    ?=3.0.0
+SEMVER_PRERELEASE ?=
+SEMVER_BUILDMETA  ?= +2
+BUILD_DATE        :=$(shell date -u -Iseconds)
+BUILD_VCS_URL     :=$(shell git config --get remote.origin.url) 
+BUILD_VCS_ID      :=$(shell git log -n 1 --date=iso-strict-local --format="%h")
+BUILD_VCS_ID_DATE :=$(shell TZ=UTC0 git log -n 1 --date=iso-strict-local --format='%ad')
+FILE_BUILD_VERSION :=$(SEMVER_VERSION)$(SEMVER_PRERELEASE)$(SEMVER_BUILDMETA)
+
+GO_LDFLAGS = -ldflags="$(BUILDFLAGS) \
+			    -X '$(GOMODULECMD).BuildVersion=$(SEMVER_VERSION)' \
+	            -X '$(GOMODULECMD).BuildPrerelease=$(SEMVER_PRERELEASE)' \
+	            -X '$(GOMODULECMD).BuildMeta=$(SEMVER_BUILDMETA)' \
+	            -X '$(GOMODULECMD).BuildDate=$(BUILD_DATE)' \
+	            -X '$(GOMODULECMD).BuildVcsUrl=$(BUILD_VCS_URL)' \
+	            -X '$(GOMODULECMD).BuildVcsId=$(BUILD_VCS_ID)' \
+		    -X '$(GOMODULECMD).BuildVcsIdDate=$(BUILD_VCS_ID_DATE)'"
+
 test_dirs=$(shell   find . -name "*_test.go" -exec dirname {} \; |  cut -d/ -f2 | sort | uniq)
 
-.PHONY: clean
+all: releases
+
+.PHONY: clean distbuild distclean linux darwin windows
 clean:
 	@echo "# cleaning autoscaler"
 	@go clean -cache -testcache
 	@rm -rf build
 
+# Releases
+releases: distclean distbuild linux darwin # windows
```

Comment:
Is this intentional to comment out windows?

Reply:
Avast antivirus was giving me a Trojan error on building the windows version.   I am still investigating, but I believe it is a false positive.

Reply:
It was a false positive.  I reported to Avast, and a recent update removed the false positive.

Reply:
If it works now, do you want to un-comment it now?